### PR TITLE
Revised architecure, rewritten on Pytorch

### PR DESCRIPTION
**Pull request by hedgegogg**
This PR keeps the same architecture, but rewrites it in PyTorch for convenience and updates the training parameters.

The .ipynb notebook is included to make the training pipeline easy to inspect and to let users train or fine-tune the model themselves. It is not intended for production use at this stage, but it helps make the project more transparent and genuinely open source.

The training data is included in model/data/poem.txt. This also is not expected to be used in production, but for now it gives a clear reference point for the model’s approximate capabilities, since those capabilities are based on the provided training data.

The notebook tracks three main training metrics:
1. Cross-Entropy Loss
This is the main optimization metric. It measures how wrong the model’s next-word probability distribution is compared with the true next word. Lower is better.

2. Next-Word Accuracy
This measures how often the model’s top predicted word exactly matches the real next word. Higher is better, but for poetry this metric is naturally strict because many different words can be reasonable.

3. Perplexity
This is derived from cross-entropy loss. It roughly means “how many plausible next words the model is confused between.” Lower perplexity means the model is more confident and better at predicting the next word.

The model is trained by minimizing cross-entropy loss. Validation loss is used for learning-rate scheduling and early stopping. Accuracy and perplexity are tracked to make training easier to interpret.

### CODE REVIEW

**Comment by hedgegogg** [comment]
Usage of `data.zip` archive
From the project root run:
`cd model && unzip -o ../data.zip`

File: model/data.zip
Comment (sachok42) [inline model/data.zip:1]: My face when I see 42 thousand strings added 😱

**Review by sachok42 — CHANGES_REQUESTED** [review]
It seems good to me, however, there is one thing you strongly need to add here and it is human readable instructions. I look into all the fourty two thousand strings and I've got no clue what should I execute here to make it all work

**Comment by hedgegogg** [comment]
Packaged the model data into `model/data.zip` and removed the extracted `model/data/` files from the PR.

The model now auto-extracts `model/data.zip` on startup if `model/data/` is missing or outdated, so users only need to pull the project and import/use the model normally. The extracted data folder is ignored by git.

**Review by iaraha — COMMENTED** [review]
Hope i don't need to check every file, and everything functions and classes are sorted in one file with instruction how to use it
